assets/wire/actions.ts

```diff
@@ -416,36 +416,65 @@ export function downloadItems(items: any) {
     return renderModal('downloadItems', {items});
 }
 
+interface DownloadPayload {
+    items: any;
+    format: any;
+    type: any;
+}
 /**
  * Start download - open download view in new window.
  *
  * @param {Array} items
  * @param {String} params
  */
 export function submitDownloadItems(items: any, params: any) {
-    return (dispatch: any, getState: any) => {
+    return async (dispatch: any, getState: any) => {
         const {format, secondaryFormat} = params;
         const userContext = context(getState());
-        let uri = `/download/${items.join(',')}?format=${format}&type=${userContext}`;
+        const payload: DownloadPayload = {
+            items,
+            format,
+            type: userContext,
+        };
+
+        let url = '/download';
         if (userContext === 'monitoring') {
             const monitoringProfile = get(getState(), 'search.activeNavigation[0]');
-            uri = `/monitoring/export/${items.join(',')}?format=${format}&monitoring_profile=${monitoringProfile}`;
-            uri = `${uri}&secondary_format=${secondaryFormat}`;
+            url = `/monitoring/export/${items.join(',')}?format=${format}&monitoring_profile=${monitoringProfile}`;
+            url = `${url}&secondary_format=${secondaryFormat}`;
+            initiateDownloadFromUrl(url);
+        }
+        else{
+            try {
+                const response = await server.post(url, payload);
+                const blob = await response.blob();
+                initiateDownloadFromBlob(blob);
+            } catch (error) {
+                console.error('Error downloading file:', error);
+            }
         }
-        browserDownload(uri);
         dispatch(setDownloadItems(items));
         dispatch(closeModal());
         analytics.multiItemEvent('download', items.map((_id: any) => getState().itemsById[_id]));
     };
 }
 
-function browserDownload(url: any) {
+function initiateDownloadFromUrl(url: string) {
     const link = document.createElement('a');
     link.download = '';
     link.href = url;
     link.click();
 }
 
+function initiateDownloadFromBlob(blob: Blob) {
+    const link = document.createElement('a');
+    link.download = '';
+    link.href = URL.createObjectURL(blob);
+    document.body.appendChild(link);
```

Comment:
is this append/remove child needed?

Reply:
I know, just wondering why it's needed here if it's not needed in the other func